proposals/records.md

```diff
@@ -38,30 +38,48 @@ The synthesized members are as follows:
+public override bool Equals(object other);
+public virtual bool Equals(T other);
+```
+`GetHashCode()` and `Equals(object other)` are overrides of the virtual methods in `System.Object`.
+Any methods on intermediate base classes that would hide those methods are ignored when overriding.
 
-* `object.GetHashCode()` override
-* `object.Equals(object)` override
-* `T Equals(T)` method, where `T` is the current type
-* `Type EqualityContract` get-only property
+Derived record types also override the `Equals(TBase other)` method from each base record type.
 
-If either `object.GetHashCode()` or `object.Equals(object)` are sealed, an error is produced.
+The record type implicitly implements `System.IEquatable<T>` where `T` is the containing type.
+Record types do not re-implement `System.IEquatable<TBase>` for any base type `TBase`.
 
-`EqualityContract` is a virtual instance property which returns `typeof(T)`. If the base type
-defines an `EqualityContract` it is overridden in the derived record. If the base `EqualityContract`
-is sealed or non-virtual, an error is produced.
+The base record class synthesizes an `EqualityContract` property. The property is overridden in
+derived record classes. The synthesized implementations return `typeof(T)` where `T` is containing type.
+```C#
+protected virtual Type EqualityContract { get; }
+```
 
-`T Equals(T)` is specified to perform value equality such that `Equals` is true if and only if
-all accessible instance fields in the receiver are equal to the fields of the parameter
-and `this.EqualityContract` equals `other.EqualityContract`.
+It is an error if the base implementations of any of the overridden members is sealed or non-virtual,
+or do not match the expected signature and accessibility.
 
-`object.Equals` performs the equivalent of
+`Equals(T other)` returns true if and only if each of the following terms are true:
+- For each field declared in the record type, the value of
+`System.Collections.Generic.EqualityComparer<TN>.Default.Equals(fieldN, other.fieldN)` where `TN` is the field type, and
+- If there is a base record type, the value of `base.Equals(other)`; otherwise
+the value of `EqualityContract.Equals(other.EqualityContract)`.
 
+The overrides of `Equals(T other)` for the base methods, including `object.Equals(object other)`, perform the equivalent of:
 ```C#
-override Equals(object o) => Equals(o as T);
+public override bool Equals(object other) => Equals(other as T);
 ```
 
+`GetHashCode()` returns the sum of the following terms where each term may be multiplied by a constant:
+- For each field declared in the record type, the value of
+`System.Collections.Generic.EqualityComparer<TN>.Default.GetHashCode(fieldN)` where `TN` is the field type, and
+- If there is a base record type, the value of `base.GetHashCode()`; otherwise
+the value of `EqualityContract.Equals(other.EqualityContract)`.
+`System.Collections.Generic.EqualityComparer<System.Type>.Default.GetHashCode(EqualityContract)`.
```

Comment:
>EqualityComparer [](start = 28, length = 16)

I think this is handled by the previous line. You shouldn't need to use an EqualityComparer to protect against null for this one.